File: test/corpus/basic/05-seeded-compatibility-test/case.go
package main

import (
	"fmt"
	"math/rand"
	colorful "github.com/lucasb-eyer/go-colorful"
)

func main() {
	fmt.Printf("=== Go Seeded FastWarmPalette Test ===\n")
	
	// Test with seed = 1
	rand.Seed(1)
	colors1 := colorful.FastWarmPalette(3)
	for i, c := range colors1 {
		fmt.Printf("Seed1-Color%d: R=%.6f G=%.6f B=%.6f Hex=%s\n", i, c.R, c.G, c.B, c.Hex())
	}
	
	fmt.Printf("\n=== Go Seeded FastHappyPalette Test ===\n")
	
	// Reset seed to 1 for happy palette
	rand.Seed(1)
	happyColors := colorful.FastHappyPalette(2)
	for i, c := range happyColors {
		fmt.Printf("Seed1-Happy%d: R=%.6f G=%.6f B=%.6f Hex=%s\n", i, c.R, c.G, c.B, c.Hex())
	}
	
	fmt.Printf("\n=== Go Seeded Test with Different Seed ===\n")
	
	// Test with seed = 42 for comparison
	rand.Seed(42)
	colors42 := colorful.FastWarmPalette(2)
	for i, c := range colors42 {
		fmt.Printf("Seed42-Color%d: R=%.6f G=%.6f B=%.6f Hex=%s\n", i, c.R, c.G, c.B, c.Hex())
	}
}

File: test/corpus/basic/05-seeded-compatibility-test/case.ts
import * as GoStyle from '../../../../src/go-style';

console.log('=== TypeScript Seeded FastWarmPalette Test ===');

// Test with seed = 1
const colors1 = GoStyle.FastWarmPaletteSeeded(3, 1);
for (let i = 0; i < colors1.length; i++) {
  const c = colors1[i];
  console.log(
    `Seed1-Color${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()}`
  );
}

console.log('\n=== TypeScript Seeded FastHappyPalette Test ===');

// Test with seed = 1 for happy palette
const happyColors = GoStyle.FastHappyPaletteSeeded(2, 1);
for (let i = 0; i < happyColors.length; i++) {
  const c = happyColors[i];
  console.log(
    `Seed1-Happy${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()}`
  );
}

console.log('\n=== TypeScript Seeded Test with Different Seed ===');

// Test with seed = 42 for comparison
const colors42 = GoStyle.FastWarmPaletteSeeded(2, 42);
for (let i = 0; i < colors42.length; i++) {
  const c = colors42[i];
  console.log(
    `Seed42-Color${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()}`
  );
}
